app/assets/javascripts/admin/addon/mixins/setting-component.js

```diff
@@ -204,10 +207,59 @@ export default Mixin.create({
     }
   },
 
+  async confirmChanges(settingKey) {
+    return await new Promise((resolve) => {
```

Comment:
now it's awaiting AND returning 🤪 

anyway, I'd be good if `dialog.alert` would return an object that had a `response` promise on it, like:

```js
const dialogInstance = await this.dialog.alert({
  buttons: [
    { label: "ok", action: () => true },
    { label: "cancel", action () => false },
  ]
});

return await dialogInstance.response;
```

Reply:
Yes that would be nicer :joy: Until then what _should_ have I done here based on Joffrey's feedback?

Reply:
Sorry can you clarify this a bit more? I am doing this:

```javascript
async confirmChanges(settingKey) {
  return new Promise((resolve) => {
    // Fallback is needed in case the setting does not have a custom confirmation
    // prompt/confirm defined.
    this.dialog.alert({
      message: I18n.t(
        `admin.site_settings.requires_confirmation_messages.${settingKey}.prompt`,
        {
          translatedFallback: I18n.t(
            "admin.site_settings.requires_confirmation_messages.default.prompt"
          ),
        }
      ),
      buttons: [
        {
          label: I18n.t(
            `admin.site_settings.requires_confirmation_messages.${settingKey}.confirm`,
            {
              translatedFallback: I18n.t(
                "admin.site_settings.requires_confirmation_messages.default.confirm"
              ),
            }
          ),
          class: "btn-primary",
          action: () => resolve(true),
        },
        {
          label: I18n.t("no_value"),
          class: "btn-default",
          action: () => resolve(false),
        },
      ],
    });
  });
},

// further down
confirm = await this.confirmChanges(key);
```

What specifically should I change without changing what `dialog.confirm` returns?

Reply:
Like, I don't know how removing `return` helps in `confirmChanges`

Reply:
Are you saying to do:

```javascript
const confirmed =  new Promise((resolve) => { //blah };
return confirmed;
```

?

Reply:
ah, sorry if you need the return value then this is the way to go:

```js
function () {
  return new Promise(…
```

i.e. no `async` and no `await`

Reply:
Ah cool, thanks!